src/components/accessories/patientExams/ExamForm/ExamForm.tsx

```diff
@@ -0,0 +1,320 @@
+    label: string;
+    align: "left" | "right" | "center" | "justify";
+  }> = [
+    { label: t("lab.resultrow"), align: "left" },
+    { label: t("lab.value"), align: "right" },
+  ];
+
+  const validationSchema = object({
+    date: string().required(t("common.required")),
+    exam: string().required(t("common.required")),
+    material: string().required(t("common.required")),
+    result:
+      currentExamProcedure === "1"
+        ? string().required(t("common.required"))
+        : string(),
+  });
+
+  const initialValues = getFromFields(fields, "value");
+  const [rowsData, setRowsData] = useState({} as { [k: string]: string });
+
+  const materialOptionsSelector = (state: IState) => {
+    if (state.laboratories.materials.data) {
+      return state.laboratories.materials.data.map((item) => {
+        return {
+          value: item,
+          label: item,
+        };
+      });
+    } else return [];
+  };
+  const materials = useSelector((state: IState) =>
+    materialOptionsSelector(state)
+  );
+
+  const examOptionsSelector = (exams: ExamDTO[] | undefined) => {
+    if (exams) {
+      return exams.map((item) => {
+        return {
+          value: item.code ?? "",
+          label:
+            (item.description &&
+              item.description?.length > 30 &&
+              item.description.slice(0, 30) + "...") ||
+            item.description + "",
+        };
+      });
+    } else return [];
+  };
+
+  const examList = useSelector((state: IState) => state.exams.examList.data);
+
+  const examRowOptionsSelector = (state: IState) => {
+    if (state.exams.examRowsByExamCode.data) {
+      return state.exams.examRowsByExamCode.data.map((item) => {
+        return {
+          value: item.description ?? "",
+          label: item.description ?? "",
+        };
+      });
+    } else return [];
+  };
+
+  const examRows = useSelector((state: IState) =>
+    examRowOptionsSelector(state)
+  );
+
+  const formik = useFormik({
+    initialValues,
+    validationSchema,
+    enableReinitialize: true,
+    onSubmit: (values) => {
+      const formattedValues = formatAllFieldValues(fields, values);
+      onSubmit(
+        formattedValues,
+        Object.values(rowsData).filter((item) => item)
+      );
+    },
+  });
+
+  const { setFieldValue, resetForm, handleBlur } = formik;
+
+  const dateFieldHandleOnChange = useCallback(
+    (fieldName: string) => (value: any) => {
+      setFieldValue(fieldName, value);
+    },
+    [setFieldValue]
+  );
+  useEffect(() => {
+    if (initialValues["exam"] !== "") {
+      setCurrentExamCode(initialValues["exam"]);
+    }
+  }, [initialValues]);
+
+  useEffect(() => {
+    if (currentExamCode) {
+      dispatch(getExamRows(currentExamCode));
+    }
+    if (currentExamCode && examList) {
+      setCurrentExamProcedure(
+        `${
+          examList?.find((item) => item.code === currentExamCode)?.procedure ??
+          ""
+        }`
+      );
+    }
```

Comment:
Why are you using this `${}` statement?

Reply:
In order to convert the result to string

Reply:
Is it not enough put an toString wrapper? (Maybe there is something that I can't see)

Reply:
OK